stdlib/public/Platform/Platform.swift

```diff
@@ -394,7 +394,7 @@ public func sem_open(
 // Some platforms don't have `extern char** environ` imported from C.
 #if os(OSX) || os(iOS) || os(watchOS) || os(tvOS) || os(FreeBSD) || os(PS4)
 public var environ: UnsafeMutablePointer<UnsafeMutablePointer<CChar>?> {
-  return _stdlib_getEnviron()
+  return _stdlib_getEnviron() ?? UnsafeMutablePointer<UnsafeMultablePointer<CChar>?>(nil)
```

Comment:
Typo, but also what? This call doesn't return a non-optional value.

Reply:
Are there really platforms where `environ` is nil?

Reply:
`_get_environ` returns `char **` which may be `NULL`.  The C standard requires that the the environment be nullable, and there are Windows environments where that is the case.

Reply:
`environ` isn't in C11 or C++14. It is in POSIX, but

> For a C-language program, an array of strings called the environment shall be made available when a process begins. The array is pointed to by the external variable _environ,_ which is defined as:
>
>     extern char **environ;

([the POSIX section on Environment Variables](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html))

and

> In addition, the following variable, which must be declared by the user if it is to be used directly:
>
>    extern char **environ;
>
> is initialized as a pointer to an array of character pointers to the environment strings. The argv and environ arrays are each terminated by a null pointer.

([the POSIX environ man page](http://pubs.opengroup.org/onlinepubs/9699919799/functions/environ.html))

and while it's possible POSIX is different, in C standardese `NULL` is not a valid "pointer to an array".

If we really care about Windows environments that aren't POSIX-compliant in this way (and maybe we do!) we should discuss this on swift-evolution, as Greg said. IUO might be the way to go, since that will be less source-breaking than a full Optional and most of the time people won't want to check for `nil`.

Reply:
I guess this is the relevant Windows info? "[_environ, _wenviron](https://docs.microsoft.com/en-us/cpp/c-runtime-library/environ-wenviron)"

> Similarly, in a program that uses `wmain`, `_environ` is initially `NULL` because the environment is composed of wide-character strings. On the first call to `_getenv` or `_putenv`, a corresponding multibyte-character string environment is created and is pointed to by `_environ`.

I'm not sure if we care about `wmain` environments from Swift at this point in time, but that does suggest an alternate solution: put a dummy `_getenv` call into the Windows implementation of `_stdlib_getEnviron`.

Reply:
Note that Darwin's implementations of `getenv` and `setenv` deliberately handle the `environ == NULL` case.

Reply:
@gparker42 glibc also supports `environ` being NULL in `getenv` and `setenv`.